csrc/ir/nodes.cpp

```diff
@@ -2063,6 +2063,34 @@ void MmaOp::setMacro(MmaMacro macro) {
   attribute<MmaMacro>(ATTR_POS_MACRO) = macro;
 }
 
+std::vector<PolymorphicValue> MmaOp::evaluate(
+    const ExpressionEvaluator& ee,
+    const std::vector<PolymorphicValue>& inputs) const {
+
+  auto& a = inputs.at(0).as<at::Tensor>();
+  auto& b = inputs.at(1).as<at::Tensor>();
+  NVF_CHECK(
+      a.dim() == b.dim(), "Either both or none of A and B should be batch");
+  NVF_CHECK(
+      a.dim() == 2 || a.dim() == 3,
+      "Must have either zero or one batch dimensions");
+
+  // Squeeze the inputs to remove the broadcasted dimensions.
+  // TODO: This is assuming the broadcast dimensions to be the last dim for the first operand
+  //      and the first dim for the second operand. If this is variable, using the inputs of the
+  //      broadcast instead of the immediate inputs may be better.
+  auto in_a = a.squeeze(-1);
```

Comment:
This is based on the broadcast dimensions in the `matmul` API on @protonu's branch. However, if it doesn't always hold, we can bypass and use the inputs to the broadcast directly.

Reply:
> This is assuming the broadcast dimensions to be the last dim for the first operand and the first dim for the second operand.

I'd explicitly `NVF_CHECK` these assumptions by examining `inputs()[0]`'s rfactor domain and `inputs()[1]`'s. 

> If this is variable, using the inputs of the broadcast instead of the immediate inputs may be better.

This would be my last resort. As Jacob said, it's more robust to avoid assuming inputs are broadcasts. Also, MmaOp, as other ops (maybe Cat is an exception?), should be standalone and have a clear semantics.